src/components/Registration/BusinessAddresses.vue

```diff
@@ -0,0 +1,412 @@
+    deliveryAddress: any
+  }
+
+  /**
+   * Whether to show the editable forms for the addresses (true)
+   * or just the static display addresses (false).
+   */
+  @Prop({ required: true })
+  readonly isEditing!: boolean
+
+  /** Whether to show errors (editing mode only). */
+  @Prop({ required: true })
+  readonly showErrors!: boolean
+
+  @Getter getRegistration!: RegistrationStateIF
+
+  @Action setRegistrationBusinessAddress!: ActionBindingIF
+
+  // BaseAddress state variables
+  // (Note that if the initial value is undefined, the class property will not be
+  // reactive, which means the changes for the properties will not be detected.)
+  private mailingAddress: AddressIF = null
+  private deliveryAddress: AddressIF = null
+  private mailingAddressValid = false
+  private deliveryAddressValid = false
+
+  // State of checkbox
+  private inheritMailingAddress = false
+  private checkboxDisabled = false
+
+  // Misc for template
+  readonly RegistrationMailingAddressSchema = RegistrationMailingAddressSchema
+  readonly RegistrationDeliveryAddressSchema = RegistrationDeliveryAddressSchema
+  readonly MAILING_ADDRESS = 'mailingAddress'
+  readonly DELIVERY_ADDRESS = 'deliveryAddress'
+
+  /** Whether the addresses form is valid. */
+  get isFormValid (): boolean {
+    return (this.mailingAddressValid && (this.deliveryAddressValid || this.inheritMailingAddress))
+  }
+
+  /** Returns true if the address object is empty or with only with default values. */
+  isEmptyAddress (address: AddressIF): boolean {
+    return (
+      isEmpty(address) ||
+      this.isSame(address, EmptyAddress) ||
+      this.isSame(address, DefaultAddress)
+    )
+  }
+
+  /** Called when component is created. */
+  created (): void {
+    // don't allow "same as" if mailing address is not in BC/Canada
+    this.checkboxDisabled = (this.mailingAddress.addressRegion !== REGION_BC ||
+      this.mailingAddress.addressCountry !== COUNTRY_CA)
+
+    // on first load, init checkbox
+    this.inheritMailingAddress = this.isSame(this.mailingAddress, this.deliveryAddress)
+  }
+
+  /** Called when the "same as" checkbox value has changed. */
+  onCheckboxChanged (): void {
+    if (this.inheritMailingAddress) {
+      // clone mailing address into delivery address
+      this.deliveryAddress = { ...this.mailingAddress }
+    } else {
+      // reset delivery address
+      this.deliveryAddress = { ...DefaultAddress }
+    }
+
+    this.setRegistrationBusinessAddress({
+      mailingAddress: this.mailingAddress,
+      deliveryAddress: this.deliveryAddress
+    })
+  }
+
+  /**
+   * Updates the specified address' data.
+   * @param addressToUpdate the address type to set the data of
+   * @param address the new address data
+   */
+  updateAddress (addressToUpdate: string, address: AddressIF): void {
+    switch (addressToUpdate) {
+      case this.MAILING_ADDRESS:
+        // avoid reactive looping
+        if (isEqual(this.mailingAddress, address)) break
```

Comment:
I found that BaseAddress component sends an update every time its address prop is updated (which is now reactive), which caused an infinite loop. This code breaks that loop.